### PR TITLE
plat-stm32mp1: fix rng driver and support secure rng services

### PR DESCRIPTION
**Pull request by etienne-lms**
This series clean stm32_rng driver to remove a bad exception masking, prevent masking interrupts during millisecond wide timeout and allow integration a secure entropy source.

### CODE REVIEW

File: core/drivers/stm32_rng.c [hunk 1]
```diff
@@ -41,6 +41,7 @@ struct stm32_rng_instance {
 	struct clk *clock;
 	unsigned int lock;
 	unsigned int refcount;
+	bool runtime;
```
Comment (jenswi-linaro) [inline core/drivers/stm32_rng.c:46]: I guess the only purpose of this is to tell `stm32_rng_release()` below if this should be released or not.
A more informative name could be `keep_post_boot` or with flipped logic `release_post_boot` or something along that.

**Review by jenswi-linaro — COMMENTED** [review]
Comments for "drivers: stm32_rng: init rng before service initcall level"

**Comment by jenswi-linaro** [comment]
For "drivers: stm32_rng: remove thread exceptions masking under spinlock"
A note on the commit message, it's not forbidden to mask exceptions it's only unnecessary under the circumstances.

**Review by jenswi-linaro — COMMENTED** [review]
Comment for "drivers: stm32_rng: remove millisecond timeout with interrupts masked"

With that fixed:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Review by jenswi-linaro — COMMENTED** [review]
For "drivers: stm32_rng: helper trace in rng path"
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Review by jenswi-linaro — COMMENTED** [review]
For "drivers: stm32_rng: register as hwrng source"
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jenswi-linaro** [comment]
For "drivers: stm32_rng: init rng before service initcall level":
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
Fixup commits squashed and review tags applied.
Thanks @jenswi-linaro for the review.

**Comment by etienne-lms** [comment]
I saw I applied your A-b tag on commit "drivers: stm32_rng: remove thread exceptions masking under spinlock" but I did not post such a tag. My bad.

**Comment by jenswi-linaro** [comment]
> I saw I applied your A-b tag on commit "drivers: stm32_rng: remove thread exceptions masking under spinlock" but I did not post such a tag. My bad.

No worries. You have it now.

**Pull request merged**